program fgetu_test;
#include( "stdlib.hhf" )

var
	Handle		:dword;
	inputF		:real80;
	
begin fgetu_test;


	// Testing fileio.getu32
	
	try
	
		stdin.getu32();
		if( eax <> 1 ) then
		
			raise(1011);
		
		endif;	
		
		stdin.getu32();
		if( eax <> 4294967295 ) then
		
			raise(1012);
		
		endif;	
				
		stdin.getu32();
		if( eax <> 2147483647 ) then
		
			raise(1014);
		
		endif;


		
		try
		
			stdin.getu32();

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		stdin.readLn();	// Clean up after ourselves.
		
		
		try
		
			stdin.getu32();

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1015 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1016 );
			
		endtry;
		stdin.readLn();
		stderr.put( "stdin.getu32 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"stdin.getu32 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;








	// Testing stdin.getu16
	
	try
	
		stdin.getu16();
		if( eax <> 0 ) then
		
			raise(1021);
		
		endif;	
		
		stdin.getu16();
		if( eax <> 1 ) then
		
			raise(1022);
		
		endif;	
		
		stdin.getu16();
		if( eax <> 65535 ) then
		
			raise(1024);
		
		endif;	
		
		stdin.getu16();
		if( eax <> 32767 ) then
		
			raise(1025);
		
		endif;
		
		try
		
			stdin.getu16();

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		stdin.readLn();	// Clean up after ourselves.
		
		
		try
		
			stdin.getu16();

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1026 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1027 );
			
		endtry;
		stdin.readLn();
		stderr.put( "stdin.getu16 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"stdin.getu16 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;





	// Testing stdin.getu8
	
	try
	
		stdin.getu8();
		if( eax <> 0 ) then
		
			raise(1031);
		
		endif;	
		
		stdin.getu8();
		if( eax <> 1 ) then
		
			raise(1032);
		
		endif;	
		
		stdin.getu8();
		if( eax <> 255 ) then
		
			raise(1033);
		
		endif;	
		
		stdin.getu8();
		if( eax <> 127 ) then
		
			raise(1035);
		
		endif;
		
		try
		
			stdin.getu8();

		  unprotected
		  
		  	// We we got to this point, we failed to raise a conversion error.
			
		  	Raise( 1005 );
			
		  exception( ex.ConversionError )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1006 );
			
		endtry;
		stdin.readLn();	// Clean up after ourselves.
		
		
		try
		
			stdin.getu8();

		  unprotected
		  
		  	// We we got to this point, we failed to raise an overflow error.
			
		  	Raise( 1036 );
			
		  exception( ex.ValueOutOfRange )
		  
		  	// Okay, we succeeded if we got this exception.
			
		  anyexception
		  
		  	// Anything else? Pass it on.
			
		  	Raise( 1037 );
			
		endtry;
		stdin.readLn();
		stderr.put( "stdin.getu8 succeeded!" nl );
		
	  anyexception
	  
		stderr.put
		( 
			nl nl 
			"***************************************************" nl
			"stdin.getu8 failed! eax=", (type uns32 eax), nl
			"Data read did not match data expected in file" nl
			"***************************************************" nl
			nl 
		);
		stdout.put( "Press Enter to Continue: " );
		stdin.readLn();
		os.exitProcess(1);
		
	endtry;




			
end fgetu_test;

													